src/components/mixins/SplashMixin.vue

```diff
@@ -54,32 +52,49 @@ export default class SplashMixin extends Vue {
         await this._getThunderstoreMods();
     }
 
-    // Get the list of Thunderstore mods from API or local cache.
+    // Get the list of Thunderstore mods from local cache or API.
     async _getThunderstoreMods() {
-        const packageListIndex = await this.fetchPackageListIndex();
-        const packageListChunks = await this.fetchPackageListChunksIfUpdated(packageListIndex);
-        this.getRequestItem('ThunderstoreDownload').setProgress(100);
-        await this.writeModsToPersistentCacheIfUpdated(packageListIndex, packageListChunks);
-        const isModListLoaded = await this.readModsToVuex();
-
-        // To proceed, the loading of the mod list should result in a non-empty list.
-        // Empty list is allowed if that's actually what the API returned.
-        // API wasn't queried at all if we already had the latest index chunk.
-        const modListHasMods = this.$store.state.tsMods.mods.length;
-        const apiReturnedEmptyList = packageListChunks && packageListChunks[0].length === 0;
-        const apiWasNotQueried = packageListIndex && packageListIndex.isLatest;
-        if (isModListLoaded && (modListHasMods || apiReturnedEmptyList || apiWasNotQueried)) {
-            await this.moveToNextScreen();
-        } else {
-            this.heroTitle = 'Failed to get the list of online mods';
-            this.loadingText = 'You may still use the manager offline, but some features might be unavailable.';
-            this.isOffline = true;
+        const hasPriorCache = await this.doesGameHaveLocalCache();
+        let hasUpdatedCache = false;
+
+        if (!hasPriorCache) {
+            const packageListIndex = await this.fetchPackageListIndex();
+            const packageListChunks = await this.fetchPackageListChunksIfUpdated(packageListIndex);
+            this.getRequestItem('ThunderstoreDownload').setProgress(100);
+            hasUpdatedCache = await this.writeModsToPersistentCacheIfUpdated(packageListIndex, packageListChunks);
         }
+
+        if (hasPriorCache || hasUpdatedCache) {
+            await this.readModsToVuex();
```

Comment:
This naming could be better by not referring to the store implementation but rather what we're doing with it.

readMods <- implies the behaviour is to be parsed, not written to.

Might be better to call it something along the lines of: `triggerStoreModListUpdate`

Reply:
Changed as suggested.